ydiff.py

```diff
@@ -799,11 +799,26 @@ def terminal_size():
         pass
     return width, height
 
+def setup_signals(entry_fn):
```

Comment:
Using decorator is a great idea, would you mind name this function `trap_interrupts`? Because it's not only for signals.